module/mfrc522_module.c

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-2.0
+
+#include <linux/module.h>
+#include <linux/init.h>
+#include <linux/spi/spi.h>
+#include <linux/regmap.h>
+#include <linux/delay.h>
+
+#include "mfrc522_spi.h"
+
+MODULE_LICENSE("GPL v2");
+MODULE_AUTHOR("ks0n");
+MODULE_DESCRIPTION("Driver for the MFRC522 RFID Chip");
+
+#define MFRC522_VERSION_BASE 0x90
+#define MFRC522_VERSION_1 0x91
+#define MFRC522_VERSION_2 0x92
+#define MFRC522_VERSION_NUM(ver) ((ver) - MFRC522_VERSION_BASE)
+
+
+static int mfrc522_spi_probe(struct spi_device *spi);
+
+static struct spi_driver mfrc522_spi = {
+    .driver = {
+        .name = "mfrc522",
+        .owner = THIS_MODULE,
+    },
+    .probe = mfrc522_spi_probe,
+};
+
+static int mfrc522_detect(struct spi_device *client) {
+    struct address_byte version_reg_read = address_byte_build(MFRC522_SPI_READ, MFRC522_VERSION_REG);
+    char version;
+
+    spi_write(client, &version_reg_read, 1);
+    mdelay(100);
+    spi_read(client, &version, 1);
+
+    switch (version) {
+            case MFRC522_VERSION_1 || MFRC522_VERSION_2:
```

Comment:
this is not rust. case expressions are constants (compile-time) this is probably not doing the intended behaviour

Reply:
Good catch ! Fixed by (f50b119)